compliance/collection/main.go

```diff
@@ -289,15 +281,23 @@ func main() {
 	if features.RHCOSNodeScanning.Enabled() {
 		rescanInterval := env.NodeRescanInterval.DurationSetting()
 		cmetrics.ObserveRescanInterval(rescanInterval, getNode())
-		log.Infof("Node Rescan interval: %s", rescanInterval.String())
+		log.Infof("Node Scanning active and configured with rescan interval %s and cache duration %s", rescanInterval.String(), env.NodeScanCacheDuration.DurationSetting())
 
+		// TODO(ROX-13935): Remove FakeNodeInventory and its FF
 		var scanner nodeinventorizer.NodeInventorizer
 		if features.UseFakeNodeInventory.Enabled() {
 			log.Infof("Using FakeNodeInventorizer")
 			scanner = &nodeinventorizer.FakeNodeInventorizer{}
 		} else {
 			log.Infof("Using NodeInventoryCollector")
-			scanner = &nodeinventorizer.NodeInventoryCollector{}
+			analyzer := &nodeinventorizer.NodeAnalyzer{}
+			scanner = nodeinventorizer.NewCachingScanner(
```

Comment:
Option: We could keep the analyzer in the if-else, and use a common scanner outside of this clause. This would mean that we would be able to cache the fake scan, but that would be a great test for the CachingScanner :)

Reply:
Sure. This is what I meant

```go
		var analyzer nodeinventorizer.NodeInventorizer
		if features.UseFakeNodeInventory.Enabled() {
			log.Infof("Using FakeNodeInventorizer")
			analyzer = &nodeinventorizer.FakeNodeInventorizer{}
		} else {
			log.Infof("Using NodeInventoryCollector")
			analyzer = &nodeinventorizer.NodeAnalyzer{}
		}
		scanner := nodeinventorizer.NewCachingScanner(
			analyzer,
			"/cache/inventory-cache",
			env.NodeScanCacheDuration.DurationSetting(),
			env.NodeScanInitialBackoff.DurationSetting(),
			env.NodeScanMaxBackoff.DurationSetting(),
			func(duration time.Duration) { time.Sleep(duration) })
		nodeInventoriesC := manageNodeScanLoop(ctx, rescanInterval, scanner)
```

Reply:
Have you maybe have a chance to give this a look/try?